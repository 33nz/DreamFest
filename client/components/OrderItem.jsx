import React from 'react'

const OrderItem = (props) => {
  const { name, quantity, id } = props.product
  return (
    <tr>
      <td>{name}</td>
      <td>
        <input
          className='update-input'
          value={quantity}
          onChange={(e) => props.update(id, e.target.value)} />
      </td>
      <td><button onClick={() => props.deleteFromOrder(id)}>
        <span className='fa fa-trash fa-2x' />
      </button></td>
    </tr>
  )
}

export default OrderItem

