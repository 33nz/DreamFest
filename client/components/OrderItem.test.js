import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import OrderItem from './OrderItem'

test('renders order item correctly', () => {
  const product = { name: 'cool product', quantity: 3 }
  const tableBody = document.createElement('tbody')
  const { asFragment } = render(<OrderItem product={product} />, {
    container: document.body.appendChild(tableBody)
  })
  expect(asFragment()).toMatchSnapshot()
})
