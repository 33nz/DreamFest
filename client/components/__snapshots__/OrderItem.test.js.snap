// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders order item correctly 1`] = `
<DocumentFragment>
  cool product3
</DocumentFragment>
`;
